src/io.c

```diff
@@ -1,4478 +1,4641 @@
+                    *fp++ = P[pindex].min_dist_to_bh;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_ACCEL:		/* acceleration */
+#ifdef OUTPUT_ACCELERATION
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k = 0; k < 3; k++) {fp[k] = All.cf_a2inv * P[pindex].GravAccel[k];}
+#ifdef PMGRID
+                    for(k = 0; k < 3; k++) {fp[k] += All.cf_a2inv * P[pindex].GravPM[k];}
+#endif
+                    if(P[pindex].Type == 0) {for(k = 0; k < 3; k++) {fp[k] += SphP[pindex].HydroAccel[k];}}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_DTENTR:		/* rate of change of internal energy */
+#ifdef OUTPUT_CHANGEOFENERGY
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = SphP[pindex].DtInternalEnergy;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_DELAYTIME:
+#ifdef GALSF_SUBGRID_WINDS
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = SphP[pindex].DelayTime;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_TSTP:		/* timestep  */
+#ifdef OUTPUT_TIMESTEP
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = GET_PARTICLE_TIMESTEP_IN_PHYSICAL(pindex);
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_BFLD:		/* magnetic field  */
+#ifdef MAGNETIC
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k=0;k<3;k++) {fp[k] = (MyOutputFloat) (Get_Gas_BField(pindex,k) * All.cf_a2inv * gizmo2gauss);}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_VDIV:		/* Divergence of Vel */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    *fp++ = P[pindex].Particle_DivVel;
+                    n++;
+                }
+            break;
+
+        case IO_VORT:		/* Vorticity */
+#if defined(TURB_DRIVING) || defined(OUTPUT_VORTICITY)
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    for(k=0;k<3;k++) {fp[k] = SphP[pindex].Vorticity[k];}
+                    fp += 3;
+                    n++;
+                }
+#endif
+            break;
+
+        case IO_SLUG_STATE_RNG:
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *ip_int++ = (rng_state_t) P[pindex].rngStateAtBirth;

```

Comment:
You need an additional level of indirection here. rngStateAtBirth is not a member of particle_data. So it should be something like P[pindex].slug_state.rngStateAtBirth instead.

Reply:
You can extract the 'lower' (first) 64 bits and the 'upper' (last) 64 bits of this 128-bit integer with:
` uint64_t low = (uint64_t)x;
  uint64_t high = (x >> 64);`
For an example of this, see: https://stackoverflow.com/questions/11656241/how-to-print-uint128-t-number-using-gcc